From c8f9ad650e7398eab662c134f87f22aa34c951bf Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Thu, 11 Dec 2025 09:49:49 +0800
Subject: [PATCH 39/48] loonggpu: use ttm_resource_manager_cleanup instead of
 manually inlining it

It should work since Linux 5.9.  And inlining an outdated copy causes a
build failure on Linux 6.19.

Signed-off-by: Xi Ruoyao <xry111@xry111.site>
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 loonggpu/loonggpu_ttm.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/loonggpu/loonggpu_ttm.c b/loonggpu/loonggpu_ttm.c
index 3fdb92f..0d9500a 100644
--- a/loonggpu/loonggpu_ttm.c
+++ b/loonggpu/loonggpu_ttm.c
@@ -1881,8 +1881,7 @@ void loonggpu_ttm_set_buffer_funcs_status(struct loonggpu_device *adev, bool ena
 		}
 	} else {
 		drm_sched_entity_destroy(&adev->mman.entity);
-		dma_fence_put(man->move);
-		man->move = NULL;
+		ttm_resource_manager_cleanup(man);
 	}
 
 	/* this just adjusts TTM size idea, which sets lpfn to the correct value */
-- 
2.52.0

